apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/newchat/NewChatSheet.kt

```diff
@@ -33,29 +41,58 @@ import kotlinx.coroutines.launch
 import kotlin.math.roundToInt
 
 @Composable
-fun NewChatSheet(chatModel: ChatModel, newChatSheetState: StateFlow<AnimatedViewState>, stopped: Boolean, closeNewChatSheet: (animated: Boolean) -> Unit) {
-  // TODO close new chat if remote host changes in model
-  if (newChatSheetState.collectAsState().value.isVisible()) BackHandler { closeNewChatSheet(true) }
-  NewChatSheetLayout(
-    newChatSheetState,
-    stopped,
-    addContact = {
-      closeNewChatSheet(false)
-      ModalManager.center.closeModals()
-      ModalManager.center.showModalCloseable { close -> NewChatView(chatModel.currentRemoteHost.value, NewChatOption.INVITE, close = close) }
-    },
-    scanPaste = {
-      closeNewChatSheet(false)
-      ModalManager.center.closeModals()
-      ModalManager.center.showModalCloseable { close -> NewChatView(chatModel.currentRemoteHost.value, NewChatOption.CONNECT, showQRCodeScanner = true, close = close) }
-    },
-    createGroup = {
-      closeNewChatSheet(false)
-      ModalManager.center.closeModals()
-      ModalManager.center.showCustomModal { close -> AddGroupView(chatModel, chatModel.currentRemoteHost.value, close) }
-    },
-    closeNewChatSheet,
-  )
+fun NewChatButton(icon: Painter, text: String, click: () -> Unit, textColor: Color = Color.Unspecified, iconColor: Color = MaterialTheme.colors.secondary, disabled: Boolean = false, extraPadding: Boolean = false) {
+  SectionItemView(click, disabled = disabled) {
+    Icon(icon, text, tint = if (disabled) MaterialTheme.colors.secondary else iconColor)
+    TextIconSpaced(extraPadding)
+    Text(text, color = if (disabled) MaterialTheme.colors.secondary else textColor)
+  }
+}
+
+@Composable
+fun NewChatOptions(addContact: () -> Unit, scanPaste: () -> Unit, createGroup: () -> Unit) {
+  val actions = remember { listOf(addContact, scanPaste, createGroup) }
+
+  Column {
+    actions.forEachIndexed { index, _ ->
+      NewChatButton(
+        icon = painterResource(icons[index]),
+        text = stringResource(titles[index]),
+        click = actions[index],
+        extraPadding = true)
+    }
+  }
+}
+@Composable
+fun ModalData.NewChatSheet(rh: RemoteHostInfo?) {
+  Column(
+    Modifier.fillMaxSize(),
+  ) {
+    Box(contentAlignment = Alignment.Center) {
+      val bottomPadding = DEFAULT_PADDING
+      AppBarTitle(
+        stringResource(MR.strings.new_chat),
+        hostDevice(rh?.remoteHostId),
+        bottomPadding = bottomPadding
+      )
+    }
```

Comment:
header jumps (has different top padding) when opening "Add contact" / "Scan / Past link" views - maybe the issue is in them but it's more noticeable now